vars/unstashParam.groovy

```diff
@@ -0,0 +1,30 @@
+import hudson.FilePath
+import hudson.model.ParametersAction
+import hudson.model.FileParameterValue
+import hudson.model.Executor
+
+def call(String name, String fname = null) {
+    def paramsAction = currentBuild.rawBuild.getAction(ParametersAction.class);
+    if (paramsAction != null) {
+        for (param in paramsAction.getParameters()) {
+            if (param.getName().equals(name)) {
+                if (! param instanceof FileParameterValue) {
+                    error "unstashParam: not a file parameter: ${name}"
+                }
+                if (env['NODE_NAME'] == null) {
+                    error "unstashParam: no node in current context"
+                }
+                if (env['WORKSPACE'] == null) {
+                    error "unstashParam: no workspace in current context"
+                }
+                nodeName = env['NODE_NAME'] == 'master' ? '(master)' : env['NODE_NAME']
+                workspace = new FilePath(Jenkins.getInstance().getComputer(nodeName).getChannel(), env['WORKSPACE'])
+                filename = fname == null ? param.getOriginalFileName() : fname
+                file = workspace.child(filename)
+                file.copyFrom(param.getFile())
```

Comment:
Aren't the uploaded files available already in the workspace? What am I missing?

Reply:
Somehow files passed by file parameter are not working without this workaround:
https://github.com/pj2/JENKINS-27413-workaround-library

Reply:
Not necessarily in this file, mind you, as its copied

Reply:
+1 if the code is from this workaround, please leave a note about it